src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java

```diff
@@ -83,50 +74,73 @@ public CsvCodec(MessageRouter<MessageGroupBatch> router, MessageRouter<EventBatc
         charset = Charset.forName(configuration.getEncoding());
     }
 
+    @NotNull
     @Override
-    public void handler(String consumerTag, MessageGroupBatch message) {
-        try {
-            List<ErrorHolder> errors = new ArrayList<>();
-            MessageGroupBatch.Builder batchBuilder = MessageGroupBatch.newBuilder();
-
-            for (MessageGroup originalGroup : message.getGroupsList()) {
-                MessageGroup.Builder groupBuilder = batchBuilder.addGroupsBuilder();
-                for (AnyMessage anyMessage : originalGroup.getMessagesList()) {
-                    if (anyMessage.hasMessage()) {
-                        groupBuilder.addMessages(anyMessage);
-                        continue;
-                    }
-                    if (!anyMessage.hasRawMessage()) {
-                        LOGGER.error("Message should either have a raw or parsed message but has nothing: {}", anyMessage);
-                        continue;
-                    }
-                    RawMessage rawMessage = anyMessage.getRawMessage();
-                    ByteString body = rawMessage.getBody();
-                    List<String[]> data = decodeValues(body);
-                    if (data.isEmpty()) {
-                        if (LOGGER.isErrorEnabled()) {
-                            LOGGER.error("The raw message does not contains any data: {}", TextFormat.shortDebugString(rawMessage));
-                        }
-                        errors.add(new ErrorHolder("The raw message does not contains any data", rawMessage));
-                        continue;
-                    }
-                    decodeCsvData(errors, groupBuilder, rawMessage, data);
+    public MessageGroup decode(@NotNull MessageGroup messageGroup) {
+        MessageGroup.Builder groupBuilder = MessageGroup.newBuilder();
+        Collection<ErrorHolder> errors = new ArrayList<>();
+        for (AnyMessage anyMessage : messageGroup.getMessagesList()) {
+            if (anyMessage.hasMessage()) {
+                groupBuilder.addMessages(anyMessage);
+                continue;
+            }
+            if (!anyMessage.hasRawMessage()) {
+                LOGGER.error("Message should either have a raw or parsed message but has nothing: {}", anyMessage);
+                continue;
+            }
```

Comment:
Please add protocol check. Codec should decode / encode message with empty or `csv` (case insensitive) protocol